src/root.component.tsx

```diff
@@ -14,29 +31,67 @@ function Root() {
     };
   }, [sidenavOpen]);
 
+  function logout() {
+    authResource
+      .logoutUser()
+      .then(() => setIsLoggedIn(false))
+      .catch(err => {
+        throw err;
```

Comment:
`.catch(err => {throw err})` is a no-op.

Calling `.catch` changes the promise from `error` to `resolved` status. But rethrowing in the callback sets it back into error status. This will cause an "unhandled promise rejection" error, which is not a good thing because not all browsers have a way of capturing and recording those.

It's good to be thinking about error handling, but we haven't established the patterns yet and this is not the right pattern. For now, you could delete this `.catch()` or you could do the following, which is closer to what the error handling will be:

```js
.catch(err => {
  // setTimeout makes this error thrown to window instead of just changing
  // the promise's status.
  setTimeout(() => {
    throw err
  })
})
```